src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Windows/Threading/Dispatcher.cs

```diff
@@ -222,10 +224,18 @@ public void VerifyAccess()
         {
             if(!CheckAccess())
             {
-                throw new InvalidOperationException(SR.Get(SRID.VerifyAccess));
+                ThrowVerifyAccess();
             }
         }
 
+        // Used to inline VerifyAccess.
+        [DoesNotReturn]
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static void ThrowVerifyAccess()
```

Comment:
This will change the behavior, I think some automated analysis tools will not work

Reply:
> some automated analysis tools will not work

They should respect the DoesNotReturn-Attribute. Or do you have something different in mind?

Reply:
@lindexi I don't think I understand what would change behavior. I may be incorrect but analysis tools should only look at the method VerifyAccess. Having private static method here only changes the analysis that may occur in this specific file. It would be great if you could explain because I don't think I understand what you mean.

Reply:
> @lindexi I don't think I understand what would change behavior. I may be incorrect but analysis tools should only look at the method VerifyAccess. Having private static method here only changes the analysis that may occur in this specific file. It would be great if you could explain because I don't think I understand what you mean.

Such as the exception call stack. Thank you.

Reply:
You don't get VerifyAccess call at the same level of stack every time, so any tool would have to go through the stack to find it, so the position should be irrelevant as long as it is there. Moreover, I think if we can get time down by 30% for something as common as VerifyAccess, it might be worth considering having the developer to fix a tool rather than have users experience slower applications.

Reply:
Sorry from the other end where exception happens it would be the same level. Still sounds like an implementation detail to me that shouldn't be relied upon.

Reply:
I really don't think anything should rely on the stacktrace of a method. It would mean that we cannot modify the content of any method because it would break the stacktrace. At this point, any change would be breaking change.

Reply:
I re-write my tools that can support the new stacktrace.

And I think it is reasonable for automated analysis tools to rely on the stacktrace.